tests/src/az_ulib_test_my_interface.c

```diff
@@ -0,0 +1,428 @@
+  }
+  AZ_ULIB_CATCH(...) {}
+
+  return AZ_ULIB_TRY_RESULT;
+}
+
+volatile long g_is_running;
+volatile long g_lock_thread;
+volatile uint32_t g_sum_sleep;
+
+static az_result my_command(az_ulib_model_in model_in, az_ulib_model_out model_out)
+{
+  const my_command_model_in* const in = (const my_command_model_in* const)model_in;
+  my_command_model_out* result = (my_command_model_out*)model_out;
+  my_command_model_in in_2;
+  uint64_t sum = 0;
+
+  (void)AZ_ULIB_PORT_ATOMIC_INC_W(&g_is_running);
+  switch (in->capability)
+  {
+    case MY_COMMAND_CAPABILITY_JUST_RETURN:
+      *result = in->return_result;
+      break;
+    case MY_COMMAND_CAPABILITY_SUM:
+      while (g_lock_thread != 0)
+      {
+        az_pal_os_sleep(10);
+      };
+      for (uint32_t i = 0; i < in->max_sum; i++)
+      {
+        if (g_sum_sleep != 0)
+        {
+          az_pal_os_sleep(g_sum_sleep);
+        }
+        sum += i;
+      }
+      *result = in->return_result;
+      break;
+    case MY_COMMAND_CAPABILITY_UNPUBLISH:
+      *result = az_ulib_ipc_unpublish(in->descriptor, in->wait_policy_ms);
+      break;
+    case MY_COMMAND_CAPABILITY_RELEASE_INTERFACE:
+      *result = az_ulib_ipc_release_interface(in->handle);
+      break;
+    case MY_COMMAND_CAPABILITY_DEINIT:
+      *result = az_ulib_ipc_deinit();
+      break;
+    case MY_COMMAND_CAPABILITY_CALL_AGAIN:
+      in_2.capability = 0;
+      in_2.return_result = AZ_OK;
+      *result = az_ulib_ipc_call(in->handle, in->command_index, &in_2, model_out);
+      break;
+    default:
+      *result = AZ_ERROR_ITEM_NOT_FOUND;
+      break;
+  }
+  (void)AZ_ULIB_PORT_ATOMIC_DEC_W(&g_is_running);
+
+  return AZ_OK;
+}
+
+static az_result my_command_span_wrapper(az_span model_in_span, az_span* model_out_span)
+{
+  AZ_ULIB_TRY
+  {
+    // Unmarshalling JSON in model_in_span to val.
+    az_json_reader jr;
+    my_command_model_in model_in = { 0 };
+    AZ_ULIB_THROW_IF_AZ_ERROR(az_json_reader_init(&jr, model_in_span, NULL));
+    AZ_ULIB_THROW_IF_AZ_ERROR(az_json_reader_next_token(&jr));
+    AZ_ULIB_THROW_IF_AZ_ERROR(az_json_reader_next_token(&jr));
+    while (jr.token.kind != AZ_JSON_TOKEN_END_OBJECT)
+    {
+      if (az_json_token_is_text_equal(
+              &jr.token, AZ_SPAN_FROM_STR(MY_INTERFACE_MY_COMMAND_CAPABILITY_NAME)))
+      {
+        AZ_ULIB_THROW_IF_AZ_ERROR(az_json_reader_next_token(&jr));
+        int32_t val;
+        AZ_ULIB_THROW_IF_AZ_ERROR(az_json_token_get_int32(&jr.token, &val));
+        model_in.capability = (uint8_t)val;
+      }
+      else if (az_json_token_is_text_equal(
+                   &jr.token, AZ_SPAN_FROM_STR(MY_INTERFACE_MY_COMMAND_MAX_SUM_NAME)))
+      {
+        AZ_ULIB_THROW_IF_AZ_ERROR(az_json_reader_next_token(&jr));
+        AZ_ULIB_THROW_IF_AZ_ERROR(az_json_token_get_uint32(&jr.token, &model_in.max_sum));
+      }
+      else if (az_json_token_is_text_equal(
+                   &jr.token, AZ_SPAN_FROM_STR(MY_INTERFACE_MY_COMMAND_DESCRIPTOR_NAME)))
```

Comment:
>MY_INTERFACE_MY_COMMAND_DESCRIPTOR_NAME [](http://example.com/codeflow?start=47&length=39)

Trow not support. In a generic platform, we cannot accept local pointer over JSON.